theano/tensor/elemwise.py

```diff
@@ -125,6 +125,7 @@ class DimShuffle(Op):
 
     _f16_ok = True
     check_input = False
+    __props__ = ("input_broadcastable", "new_order")
 
     def __init__(self, input_broadcastable, new_order, inplace=False):
```

Comment:
For the full change about Dimshuffle inplace, you should change the default to True. Then if it is false, raise an error. Also, update the docstring of this class. In this file, there is a few places, where we set inplace=False. Just remove that paremeter when instanciating a Dimshuffle.